File: 412-fizz-buzz/412-fizz-buzz.java
class Solution {
    public List<String> fizzBuzz(int n) {
      List<String> res = new ArrayList<>();  
      for(int i=1;i<=n;i++)
        {
          boolean divBy3=(i%3==0);
          boolean divBy5=(i%5==0);
          if(divBy3 && divBy5)
          {
            res.add("FizzBuzz");
          }
          else if(divBy3)
          {
            res.add("Fizz");
          }
          else if(divBy5)
          {
            res.add("Buzz");
          }
          else res.add(String.valueOf(i));
        }
      return res;
    }
}

File: 412-fizz-buzz/412-fizz-buzz.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {

   let i =1;
  const res = []; 
  
  while(i<=n)
  {
    var divBy3 = false;
    var divBy5 = false;
    
    if(i%3===0) divBy3 = true;
    if(i%5===0) divBy5 = true;

    if(divBy3 && divBy5) res.push("FizzBuzz");
    else if(divBy3) res.push("Fizz");
    else if(divBy5) res.push("Buzz");
    else res.push(i.toString());
    i++;
  }
 return res;
};